20.7 : Code modifications and improvements

FJBDev opened the pull request:
- SelectInBetweenTimeSlices : Finalization per code review 
    * Implementation of timerExec() version
    * Reading the pref field in the tour editor and removed the additonal parameter isSelectInBetweenTimeSlices in SelectionChartXSliderPosition
    * Tooltip line wrapping

</br>

- Easy Import Config Dialog:  Displaying the file system icon for a better user experience

![image](https://user-images.githubusercontent.com/25235665/86820182-ebd36e80-c045-11ea-94bd-c48eee895f6f.png)


![image](https://user-images.githubusercontent.com/25235665/86820252-0279c580-c046-11ea-9e39-08e7da06da1c.png)


</br>

- Formatting and typo fixes

- Suunto Quest XML small improvements

- Adding the ability to bulk edit the weather conditions

![image](https://user-images.githubusercontent.com/25235665/86978750-dbea8600-c13c-11ea-8673-01728de88b47.png)


</br>

- Readme update

wolfgang-ch commented on bundles/net.tourbook/src/net/tourbook/ui/views/tourDataEditor/TourDataEditorView.java:
```diff
@@ -7890,12 +7901,25 @@ private void selectTimeSlice(final SelectionChartInfo chartInfo) {
+      if (_prefStore.getBoolean(
+            ITourbookPreferences.TOGGLE_STATE_SELECT_INBETWEEN_TIME_SLICES)) {
+
+         // delay the selection of multiple lines, moving the mouse can occur very often
+         _parent.getDisplay().timerExec(50, new Runnable() {
+
+            @Override
+            public void run() {
+
+               if (_parent.isDisposed()) {
+                  return;
+               }
+
+               final int minSelectedValue = Math.min(chartInfo.leftSliderValuesIndex, chartInfo.rightSliderValuesIndex);
+               final int maxSelectedValue = Math.max(chartInfo.leftSliderValuesIndex, chartInfo.rightSliderValuesIndex);
+               table.setSelection(minSelectedValue, maxSelectedValue);
+            }
+         });
```
The only reason why a delayed execution is performed, is not applied from https://github.com/wolfgang-ch/mytourbook/pull/192#discussion_r443194322 
-> The delayed execution should skip all selection events which occurred in the last 50 ms, therefor the `__runningId` is set and checked

FJBDev commented:
>>> -> The delayed execution should skip all selection events which occurred in the last 50 ms, therefor the __runningId is set and checked

@wolfgang-ch Fixed in my last commit [e921e55 ](https://github.com/wolfgang-ch/mytourbook/pull/199/commits/e921e55db1b335d5df122f7ddb05e1d6836f8d77)

wolfgang-ch commented on bundles/net.tourbook/icon-factory/common.svg:
```diff
@@ -7510,5 +7510,14 @@
+       class="cls-1"
+       id="path6"
+       style="stroke-width:0.04622141"
+       inkscape:export-filename="C:\Users\frederic\git\MT\mytourbook\bundles\net.tourbook\icons\hard-drive.png"
```
The hard drive image is a static image which is put into common.svg and then exported. All svg files MUST contain only self painted images to have no legal trouble.

![hard-drive-image](https://user-images.githubusercontent.com/1283445/87217918-af0dbd00-c34e-11ea-9dbe-9ecc134ff921.png)

FJBDev replied:
I have obtained this SVG via _Wikimedia Commons, the free media repository_

Here is the [link](https://commons.wikimedia.org/wiki/File:Hard-drive.svg) where you can see that it is legal to use this SVG.

But if you prefer, I can draw one myself, let me know

wolfgang-ch commented:
I removed the hard drive image and replaced it with a dummy image until the image origin is verified

wolfgang-ch commented:
The use of the dropbox icon in MT is for me also questionable, when I google for this icon I find in https://commons.wikimedia.org/wiki/File:Dropbox_Icon.svg this text

`If you want to use it, you have to ensure that you have the legal right to do so and that you do not infringe any trademark rights`

FJBDev commented:
> 
> 
> The use of the dropbox icon in MT is for me also questionable, when I google for this icon I find in https://commons.wikimedia.org/wiki/File:Dropbox_Icon.svg this text
> 
> `If you want to use it, you have to ensure that you have the legal right to do so and that you do not infringe any trademark rights`

I have obtained the Dropbox logo SVG from the official Dropbox website, so it is legal

[Source](https://www.dropbox.com/branding)
Please reference this folder for the following.

_The Dropbox logo is a universal signature we use across all our communications. We want it to be instantly recognizable, so consistency is important—please don’t edit, change, distort, recolor, or reconfigure it._

SVG source:
https://www.dropbox.com/sh/85o3g5daqgy91wy/AACY4Z73YMY9jOwSDGSrrCvda/Dropbox/SVG?dl=0&preview=Dropbox.svg&subfolder_nav_tracking=1

FJBDev commented:
> 
> 
> I removed the hard drive image and replaced it with a dummy image until the image origin is verified

If you prefer, I can also use this icon as it looks like a harddrive and it has the same look as the other icons in MT

https://github.com/wolfgang-ch/mytourbook/blob/main/bundles/net.tourbook/icons/database.png

![image](https://user-images.githubusercontent.com/25235665/87227338-8fc75d80-c357-11ea-8a85-855d832756d3.png)

Let me know

wolfgang-ch commented:
OK, all is legal :-)

The reason why I came up with my thoughts, I wanted to test the hard drive image a bit uncluttered, without the surrounding border because I think this image is not inteded to have a small size of 16x16. So I opened the svg file and saw it as an imported file which I could not ungroup.

I also discovered that the line border size is not exactly on the pixel border which should be used to have a clean representation.

So I would be happy when you come up with an uncluttered hard drive icon :-)

I think when an external svg image is used, it sould be put into another svg file, to separate the self painted icons and external icons.

FJBDev commented:
@wolfgang-ch  I have created my own hard drive logo with Inkscape :-)

See my latest [PR](https://github.com/wolfgang-ch/mytourbook/pull/202)
